fideslib/db/base_class.py

```diff
@@ -0,0 +1,302 @@
+        """Returns the names of all nullable fields on the wrapped model"""
+        return [
+            field.name
+            for field in list(cls.__table__.columns)  # type: ignore # pylint: disable=E1101
+            if field.nullable
+        ]
+
+    @classmethod
+    def get(cls, db: Session, *, id: Any) -> FidesBase | None:  # pylint: disable=W0622
+        """Fetch a database record via a table ID"""
+        return db.query(cls).get(id)
+
+    @classmethod
+    def get_by(
+        cls,
+        db: Session,
+        *,
+        field: str,
+        value: str | int,
+    ) -> list[FidesBase]:
+        """Fetch a database record via a dynamic key, supplied at call time"""
+        kwargs = {field: value}
+        return db.query(cls).filter_by(**kwargs).first()  # type: ignore
+
+    @classmethod
+    def query(cls, db: Session) -> Query:
+        """Create a blank query for the class"""
+        return db.query(cls)
+
+    @classmethod
+    def all(cls, db: Session) -> list[OrmWrappedFidesBase]:  # pylint: disable=W0622
+        """Fetch all database records in table"""
+        return db.query(cls).all()
+
+    @classmethod
+    def filter(
+        cls,
+        db: Session,
+        *,
+        conditions: Any,  # TODO: What pydantic types are these? # pylint: disable=W0511
+    ) -> Query:
+        """Fetch multiple models from a database table"""
+        return db.query(cls).filter(conditions)
+
+    @classmethod
+    def create(cls, db: Session, *, data: dict[str, Any]) -> FidesBase:
+        """Create a new row in the database"""
+        # Build properly formatted key for applicable classes
+        if hasattr(cls, "key"):
+            data["key"] = get_key_from_data(data, cls.__name__)
+            if db.query(cls).filter_by(key=data["key"]).first():
+                raise KeyOrNameAlreadyExists(
+                    f"Key {data['key']} already exists in {cls.__name__}. Keys will be snake-cased names if not provided. "  # pylint: disable=C0301
+                    f"If you are seeing this error without providing a key, please provide a key or a different name."  # pylint: disable=C0301
+                    ""
+                )
+
+        if hasattr(cls, "name"):
+            data["name"] = data.get("name", None)
+            if db.query(cls).filter_by(name=data["name"]).first():
+                raise KeyOrNameAlreadyExists(
+                    f"Name {data['name']} already exists in {cls.__name__}."
+                )
+
+        # Create
+        db_obj = cls(**data)  # type: ignore
+        return cls.persist_obj(db, db_obj)
+
+    @classmethod
+    def get_by_key_or_id(
+        cls, db: Session, *, data: dict[str, Any]
+    ) -> FidesBase | list[FidesBase] | None:
+        """Retrieves db object by id, if provided, otherwise attempts by key"""
+        db_obj: FidesBase | list[FidesBase] | None = None
+        if data.get("id") is not None:
+            # If `id` has been included in `data`, preference that
```

Comment:
IMO this comment is redundant given the simplicity of the preceding conditional. There are several other comments in this file that should also be removed.